scripts/get_nn_models.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+# This file is part of ap_verify_dataset_template.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Script for ingesting pretrained neural net models for this dataset.
+
+Running this script allows for updates or replacements of the existing model.
+
+Example:
+$ python get_nn_models.py -m rbResnet50-DC2
+imports rbResnet50-DC2 from /repo/main to this dataset's preloaded repo. See
+get_nn_models.py -h for more options.
+"""
+
+import argparse
+import logging
+import os
+import sys
+
+import lsst.log
+import lsst.skymap
+from lsst.daf.butler import Butler, CollectionType
+
+logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+lsst.log.configure_pylog_MDC("DEBUG", MDC_class=None)
+
+
+# Normal model usage does not use chains, but need them here both to decouple
+# this script from make_all.sh and to hook in to ap_verify's automated
+# pipeline execution.
+MODEL_PREFIX = "pretrained_models"  # Hardcoded into meas_transiNet, do not change
```

Comment:
I did consider that, but that would mean that maintaining (not just running) the repository would require `meas_transiNet` to be set up. That struck me as a bit error prone in a way that requiring `daf_butler` is not.

Reply:
I didn't quite follow the down-side you are referring to, and am still leaning towards the non-hard-coded implementation. But will leave the final choice to you.

Reply:
I'll change it.

Reply:
Ah! Now I remember why I didn't like it: because `StorageAdapterButler` is a package-private class. To use `packages_parent_collection` you need to qualify it all the way: `lsst.meas.transiNet.modelPackages.storageAdapterButler.StorageAdapterButler.packages_parent_collection`.